src/alert-fastlog.c

```diff
@@ -141,24 +141,30 @@ int AlertFastLogger(ThreadVars *tv, void *data, const Packet *p)
             action = "[wDrop] ";
         }
 
-        char proto[16] = "";
+        /* Create the alert string without locking. */
+        int size = 0;
         if (likely(decoder_event == 0)) {
+            char proto[16] = "";
             if (SCProtoNameValid(IP_GET_IPPROTO(p)) == TRUE) {
                 strlcpy(proto, known_proto[IP_GET_IPPROTO(p)], sizeof(proto));
             } else {
                 snprintf(proto, sizeof(proto), "PROTO:%03" PRIu32, IP_GET_IPPROTO(p));
             }
-        }
-
-        /* Create the alert string without locking. */
-        int size = 0;
-        if (likely(decoder_event == 0)) {
-            PrintBufferData(alert_buffer, &size, MAX_FASTLOG_ALERT_SIZE, 
+            char srcPortOrIcmp[17] = "";
+            char dstPortOrIcmp[17] = "";
+            if (IP_GET_IPPROTO(p) == IPPROTO_ICMP) {
+                snprintf(srcPortOrIcmp, sizeof(srcPortOrIcmp), "type=%03" PRIu8 "code=%03" PRIu8, p->icmp_s.type, p->icmp_s.code);
```

Comment:
Should `"code=%03"` be `" code=%03"`?

Reply:
If we do that scripts processing fast.log with `cut` will fail...
Not sure what the best option is...

Reply:
This was broken in c662383b53f059dc221d74841228adda2c9a200f. Before that sp was a union with type, dp with code. So the output simply used:
```
<ip>:<type> -> <ip>:<code>
1.2.3.4:8 -> 5.6.7.8:0
```

Reply:
Ok, so icmp has always same type and code for source and destination ?